### PR TITLE
Allow a custom env for more `xcrun` invocations

### PR DESCRIPTION
**Pull request by neonichu**
Basically a continuation of #1766

### CODE REVIEW

**Comment by aciidb0mb3r** [comment]
We shouldn't put the env in UserToolchain instead of getClangCompiler()

**Comment by neonichu** [comment]
I was 🤔 about that, but decided that there might be a need to have a different environment for manifest parsing vs. building. I guess if someone really needs that customisability, they can create a custom `Toolchain` type though. I'll update the PR 🔨.

**Comment by neonichu** [comment]
@swift-ci please smoke test

File: Sources/Workspace/UserToolchain.swift [hunk 1]
```diff
@@ -129,6 +129,8 @@ public final class UserToolchain: Toolchain {
         return lookupExecutablePath(filename: getenv(variable), searchPaths: searchPaths)
     }
 
+    private let environment: [String:String]
```
Comment (aciidb0mb3r) [inline Sources/Workspace/UserToolchain.swift]: nit: needs doc comment and space between `:` and `String`.

File: Sources/Workspace/UserToolchain.swift [hunk 2]
```diff
@@ -159,6 +161,7 @@ public final class UserToolchain: Toolchain {
 
     public init(destination: Destination, environment: [String:String] = Process.env) throws {
```
Comment (aciidb0mb3r) [inline Sources/Workspace/UserToolchain.swift]: nit: `/s/[String:String/[String: String]/`

File: Sources/Workspace/UserToolchain.swift (hunk 1 above)
Comment (aciidb0mb3r) [reply Sources/Workspace/UserToolchain.swift]: should we call it `processEnv` maybe?

**Comment by neonichu** [comment]
@swift-ci please smoke test

**Comment by neonichu** [comment]
@swift-ci please smoke test

**Comment by neonichu** [comment]
This should be good to merge now.

**Pull request merged**